client/state/plugins/installed/actions.js

```diff
@@ -278,7 +297,10 @@ export function updatePlugin( siteId, plugin ) {
 			pluginId,
 		};
 
-		if ( ! plugin?.update || plugin?.update?.recentlyUpdated ) {
+		if (
+			! pluginHasTruthySiteProp( 'update', plugin, siteId ) ||
+			( siteId && plugin?.sites?.[ siteId ]?.update?.recentlyUpdated )
```

Comment:
This is the only check which breaks the pattern since it is looking for `plugin.update.recentlyUpdated`. I figured it was better to just have this exception than to complicate `pluginHasTruthySiteProp` by making it handle this case as well.